crates/router/src/core/errors/api_error_response.rs

```diff
@@ -254,3 +255,168 @@ impl actix_web::ResponseError for ApiErrorResponse {
+                AER::ForbiddenPrivateResource(ApiError::new("HE", 2, "Refund does not exist in our records.", None))
+            }
+            Self::CustomerNotFound => {
+                AER::ForbiddenPrivateResource(ApiError::new("HE", 2, "Customer does not exist in our records", None))
+            }
+            Self::ConfigNotFound => {
+                AER::ForbiddenPrivateResource(ApiError::new("HE", 2, "Config key does not exist in our records.", None))
+            }
+            Self::PaymentNotFound => {
+                AER::ForbiddenPrivateResource(ApiError::new("HE", 2, "Payment does not exist in our records", None))
+            }
+            Self::PaymentMethodNotFound => {
+                AER::ForbiddenPrivateResource(ApiError::new("HE", 2, "Payment method does not exist in our records", None))
+            }
+            Self::MerchantAccountNotFound => {
+                AER::ForbiddenPrivateResource(ApiError::new("HE", 2, "Merchant account does not exist in our records", None))
+            }
+            Self::MerchantConnectorAccountNotFound => {
+                AER::ForbiddenPrivateResource(ApiError::new("HE", 2, "Merchant connector account does not exist in our records", None))
+            }
+            Self::ResourceIdNotFound => {
+                AER::ForbiddenPrivateResource(ApiError::new("HE", 2, "Resource ID does not exist in our records", None))
+            }
+            Self::MandateNotFound => {
+                AER::ForbiddenPrivateResource(ApiError::new("HE", 2, "Mandate does not exist in our records", None))
+            }
+            Self::ReturnUrlUnavailable => AER::NotFound(ApiError::new("HE", 3, "Return URL is not configured and not passed in payments request", None)),
+            Self::RefundNotPossible { connector } => {
+                AER::BadRequest(ApiError::new("HE", 3, "This refund is not possible through Hyperswitch. Please raise the refund through {connector} dashboard", None))
+            }
+            Self::MandateValidationFailed { reason } => {
+                AER::BadRequest(ApiError::new("HE", 3, "Mandate Validation Failed", Some(Extra { reason: Some(reason.clone()), ..Default::default() })))
+            }
+            Self::PaymentNotSucceeded => AER::BadRequest(ApiError::new("HE", 3, "The payment has not succeeded yet. Please pass a successful payment to initiate refund", None)),
+            Self::SuccessfulPaymentNotFound => {
+                AER::ForbiddenPrivateResource(ApiError::new("HE", 4, "Successful payment not found for the given payment id", None))
+            }
+            Self::IncorrectConnectorNameGiven => {
+                AER::ForbiddenPrivateResource(ApiError::new("HE", 4, "The connector provided in the request is incorrect or not available", None))
+            }
+            Self::AddressNotFound => {
+                AER::ForbiddenPrivateResource(ApiError::new("HE", 4, "Address does not exist in our records", None))
```

Comment:
Shouldn't this be a normal NOT_FOUND instead of `ForbiddenPrivateResource`,

Those were intended for resources which exist but the user does not have access to them...